search-mapper/src/main/java/org/hibernate/search/annotations/IndexedEmbedded.java

```diff
@@ -77,31 +77,29 @@ public void process(PropertyMappingStep mappingContext, IndexedEmbedded annotati
             cleanedUpPrefix = cleanedUpPrefix.substring(1);
          }
 
-         if ( cleanedUpPrefix.isEmpty() ) {
+         if (cleanedUpPrefix.isEmpty()) {
             cleanedUpPrefix = null;
          }
 
          Integer cleanedUpMaxDepth = annotation.depth();
-         if ( cleanedUpMaxDepth.equals( -1 ) ) {
+         if (cleanedUpMaxDepth.equals(-1)) {
             cleanedUpMaxDepth = null;
          }
 
          String[] includePathsArray = annotation.includePaths();
          Set<String> cleanedUpIncludePaths;
-         if ( includePathsArray.length > 0 ) {
+         if (includePathsArray.length > 0) {
             cleanedUpIncludePaths = new HashSet<>();
-            Collections.addAll( cleanedUpIncludePaths, includePathsArray );
-         }
-         else {
+            Collections.addAll(cleanedUpIncludePaths, includePathsArray);
+         } else {
             cleanedUpIncludePaths = Collections.emptySet();
          }
 
-         mappingContext.indexedEmbedded()
-               .extractors( ContainerExtractorPath.defaultExtractors() )
-               .prefix( cleanedUpPrefix )
```

Comment:
@gustavonalle FYI replacing `.indexedEmbedded().prefix( cleanedUpPrefix )` with `.indexedEmbedded(cleanedUpPrefix)` implies a change in behavior.... Essentially the parameter of `indexedEmbedded` is a *name*, but the prefix really is really just a prefix prepended to included fields.

For example if you include field `myField` and want it to be registered at path `myEmbedded.myField`, you'll have to use `.indexedEmbedded("myEmbedded")` or `.indexedEmbedded().prefix("myEmbedded.")` (notice the trailing dot). If you call `.indexedEmbedded("myEmbedded.") you'll get an error.

Another example, if you include field  `myField` and want it to be registered at path `myEmbedded_myField`, you'll have to use `.indexedEmbedded().prefix("myEmbedded_")`. You cannot achieve anything similar with `.indexedEmbedded(<some name>)`, since calling `.indexedEmbedded("myEmbedded_")` will result in the path `myEmbedded_.myField`.

I'm not sure you wanted that?

Reply:
I am puzzled why no tests failed because of this change. 
And I didn't want to necessarily to change this behaviour, I noticed ```.indexedEmbedded().prefix( cleanedUpPrefix )``` was deprecated, and I thought the replacement would be using a name according to ```org.hibernate.search.mapper.pojo.mapping.definition.programmatic.PropertyMappingIndexedEmbeddedStep#prefix``

Reply:
I will send a follow up PR to fix this